Mage/src/main/java/mage/abilities/effects/common/DamageTargetEffect.java

```diff
@@ -189,17 +208,37 @@ public String getText(Mode mode) {
                 sb.append("that target");
             }
         }
-        if (!message.isEmpty()) {
-            if (message.equals("1")) {
-                sb.append(" equal to the number of ");
-            } else {
-                if (message.startsWith("the") || message.startsWith("that") || message.startsWith("twice")) {
-                    sb.append(" equal to ");
+        if (phrasing == ValuePhrasing.LEGACY){
+            if (!message.isEmpty()) {
+                if (message.equals("1")) {
+                    sb.append(" equal to the number of ");
```

Comment:
Is that code really generate same texts? Looks like old/legacy code used much more text variants with “number of” and “twice” words:

![IMG_0739](https://github.com/user-attachments/assets/3ae2441e-7d5f-4f89-a13f-631cd6fd4430)

Reply:
yes, i'm verifying this PR and future PRs on this rework by diffing a `VerifyCardDataTest` with `FULL_ABILITIES_CHECK_SET_CODES = "*"`. Seevral oracle text errors have been fixed, but all of them have to do with the changes around `getBoostText`, with the exception of [[Foundry Champion]], which i updated using what is now `withTextPhrasing`.

Reply:
Do not trust total stats. It can be false positive. If you fixed a text in one popular use case (1000 cards) but broke dozen other not popular use cases (100 cards) then total stats will be +900. And you will never to know about it by stats.

Possible solution: track text stats history after each rule/effect gen change.

Reply:
The way I do it is by comparing diffs, yes.